### PR TITLE
Fix some build warnings caused by unused variables/unnecessary tests

### PR DESCRIPTION
**Pull request by msimberg**

### CODE REVIEW

File: tests/unit/lcos/when_each.cpp [hunk 1]
```diff
@@ -40,7 +40,6 @@ void test_when_each_from_list()
 
             unsigned id = fut.get();
 
-            HPX_TEST(id >= 0);
```
Comment (sithhell) [inline tests/unit/lcos/when_each.cpp]: this should rather be `HPX_TEST(id > 0)`, shouldn't it?

File: tests/unit/lcos/when_each.cpp [hunk 2]
```diff
@@ -52,7 +51,6 @@ void test_when_each_from_list()
             unsigned id = fut.get();
 
             HPX_TEST_EQ(idx, id);
-            HPX_TEST(id >= 0);
```
Comment (sithhell) [inline tests/unit/lcos/when_each.cpp]: Same as above

File: tests/unit/lcos/when_each.cpp [hunk 3]
```diff
@@ -101,7 +99,6 @@ void test_when_each_from_list_iterators()
 
             unsigned id = fut.get();
 
-            HPX_TEST(id >= 0);
```
Comment (sithhell) [inline tests/unit/lcos/when_each.cpp]: And many more to come...

File: tests/unit/lcos/when_each.cpp (hunk 1 above)
Comment (msimberg) [reply tests/unit/lcos/when_each.cpp]: If I understand the test correctly, no. At least `id` will be 0 the for the first future processed by `when_each` a bit further down. The important test is `HPX_TEST_EQ(id, idx)`.

Comment (sithhell) [reply tests/unit/lcos/when_each.cpp]: Ahh, I see now. Good.

**Pull request merged**